src/tools/perf/lib/ucp_tests.cc

```diff
@@ -90,16 +90,36 @@ class ucp_perf_test_runner {
         }
     }
 
+    void make_blocking_progress() {
+        int count = 0;
+
+        while (ucp_worker_progress(m_perf.ucp.worker)) {
```

Comment:
while (ucp_worker_progress(m_perf.ucp.worker) == 0)

Reply:
@yosefe `make_blocking_progress` aims to drain the worker and call worker_wait if the worker is already drained (if `count == 0`). From the documentation of worker_wait, it looks like a requirement to drain worker before calling worker_wait. Wouldn't `while(ucp_worker_progress() == 0)` have the effect of polling?

Reply:
sorry, i meant != 0

Reply:
@yosefe Looks like gtests hang if I use the following code:
```
while (ucp_worker_progress(m_perf.ucp.worker) != 0) {
}

ucp_worker_wait(m_perf.ucp.worker);
```
By forcing worker to drain completely and necessarily calling worker_wait, we are preventing either the requestor or responder from performing other transfers while being blocked in ucp_worker_wait. If the same thread is expected to perform the roles of both requestor and responder then draining worker and calling wait may result in a deadlock if the wakeup is also provided by the same thread, right?

Reply:
how about
```
if (ucp_worker_progress() == 0) {
   ucp_worker_wait(m_perf.ucp.worker);
}
```